lib/rules/no-restricted-imports.js

```diff
@@ -333,6 +398,25 @@ module.exports = {
 		 * @private
 		 */
 		function checkRestrictedPathAndReport(importSource, importNames, node) {
+			// Skip if this is a type-only import and it's allowed
+			if (
+				(node.type === "ImportDeclaration" ||
+					node.type === "TSImportEqualsDeclaration") &&
+				isTypeOnlyImport(node) &&
+				isAllowedTypeImportPath(importSource)
+			) {
+				return;
+			}
+
+			// Skip if this is a type-only export and it's allowed
+			if (
+				node.type === "ExportNamedDeclaration" &&
+				isTypeOnlyExport(node) &&
+				isAllowedTypeImportPath(importSource)
```

Comment:
[Testing] Removing this `&& isAllowedTypeImportPath(importSource)` didn't fail any unit tests. I added one locally that was a copy of the last test, `// Both regular and type imports should still be restricted`, but switched to exports instead of imports.

Reply:
Added